src/middlewares/check_auth.rs

```diff
@@ -228,14 +228,13 @@ mod tests {
         when!(mocked_users_cache.add_user).times(0);
         when!(mocked_synapse.who_am_i).times(0);
 
-        let mocked_components = CustomComponents {
-            synapse: Some(mocked_synapse),
-            db: Some(mocked_db),
-            users_cache: Some(mocked_users_cache),
-            redis: Some(mocked_redis),
-        };
+        let mut mocked_comps = CustomComponents::builder();
+        mocked_comps.with_synapse(mocked_synapse);
+        mocked_comps.with_db(mocked_db);
+        mocked_comps.with_redis(mocked_redis);
+        mocked_comps.with_users_cache(mocked_users_cache);
```

Comment:
This works but I was thinking of something [more like this](https://rust-unofficial.github.io/patterns/patterns/creational/builder.html).

This way you could have chains of the kind:
```rust
let mocked_comps: CustomComponents = CustomComponents::builder()
        .with_synapse(mocked_synapse)
        .with_db(mocked_db)
        .with_redis(mocked_redis)
        .with_users_cache(mocked_users_cache)
        .build();
```
Or in the last case scenario:
```rust
let mocked_comps: CustomComponents = CustomComponents::builder()
        .with_synapse(mocked_synapse)
        .build();
```
The builder itself being a separate `CustomComponentsBuilder` type itself.